script.js

```diff
@@ -1,4 +1,216 @@
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  var myOutput;
+  // When player click submit to start the game
+  if (gameMode == start_game) {
+  //Assign cards to the players
+  var playerCard = deck.pop();
+  var compCard = deck.pop();
+  // console.log(playerCard.rank);
+  // console.log(compCard.rank);
+  player.push(playerCard.rank);
+  comp.push(compCard.rank);
+  console.log(`player array: ${player}`);
+  console.log(`comp array: ${comp}`);
+  //Sum up both player card
+  totalPlayerSum = sumCard (player);
+  totalCompSum = sumCard (comp);
+  console.log (`total player sum ${totalPlayerSum}`);
+  console.log (`total comp sum: ${totalCompSum}`); 
+  playerDisplay.push(playerCard.cardName);
+  playerDisplay.push(playerCard.suit);
+  console.log(`player display ${playerDisplay}`);
+  compDisplay.push(compCard.cardName);
+  compDisplay.push(compCard.suit);
+  console.log(`comp display ${compDisplay}`);
+  console.log("index" + index);
+  if (totalPlayerSum == '21'){
+    return 'Player Win';
+  }
+  if (totalCompSum == '21'){
+    return 'Comp Win';
+  }
+  if (index == 1){
+    gameMode = hit_or_stand;
+    //console.log(`hi`);
+    return `Player cards: ${playerDisplay}<br>Player sum: ${totalPlayerSum}<br><br> Comp cards: ${compDisplay}<br> Comp sum: ${totalCompSum}<br><br> Please enter 'hit' or 'stand'`;
+  }
+  index = index + 1;
+  myOutput = `Player cards: ${playerDisplay}<br>Player sum: ${totalPlayerSum}<br><br> Comp cards: ${compDisplay}<br> Comp sum: ${totalCompSum}<br><br> Please click submit again.`
+  }
+  if (gameMode == hit_or_stand){
+    if ((input == 'hit') && (totalCompSum < '17' )){
+      //Assign cards to the players
+      var playerCard = deck.pop();
+      var compCard = deck.pop();
+      // console.log(playerCard.rank);
+      // console.log(compCard.rank);
+      player.push(playerCard.rank);
+      comp.push(compCard.rank);
+      // console.log(`player array: ${player}`);
+      // console.log(`comp array: ${comp}`);
+      //Sum up both player card
+      totalPlayerSum = sumCard(player);
+      totalCompSum = sumCard(comp);
+      // console.log(`total player sum ${totalPlayerSum}`);
+      // console.log(`total comp sum: ${totalCompSum}`);
+      playerDisplay.push(playerCard.cardName);
+      playerDisplay.push(playerCard.suit);
+      //console.log(`player display ${playerDisplay}`);
+      compDisplay.push(compCard.cardName);
+      compDisplay.push(compCard.suit);
+      // console.log(`comp display ${compDisplay}`);
+      // console.log("index" + index);
+      //console.log(`hi`);
+      if (totalPlayerSum == "21") {
+        return "Player Win";
+      }
+      if (totalCompSum == "21") {
+        return "Comp Win";
+      }
+      myOutput = `Player cards: ${playerDisplay}<br>Player sum: ${totalPlayerSum}<br><br> Comp cards: ${compDisplay}<br> Comp sum: ${totalCompSum}<br><br> Please click submit again.`;
+    }
+    if (input == 'stand'){
```

Comment:
add a provision if input != to hit or stand. It comes out as undefined and the code can not run properly